core/keyserver/keyserver.go

```diff
@@ -276,7 +277,93 @@ func (s *Server) ListEntryHistory(ctx context.Context, in *pb.ListEntryHistoryRe
 // ListUserRevisions returns a list of revisions covering a period of time.
 func (s *Server) ListUserRevisions(ctx context.Context, in *pb.ListUserRevisionsRequest) (
 	*pb.ListUserRevisionsResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	offset := int64(0)
+	lastVerified := in.GetLastVerifiedTreeSize()
+	if in.GetPageToken() != "" {
+		token := &rtpb.ListUserRevisionsToken{}
+		if err := DecodeToken(in.GetPageToken(), token); err != nil {
+			glog.Errorf("invalid page token %v: %v", in.GetPageToken(), err)
+			return nil, status.Errorf(codes.InvalidArgument, "Invalid page_token provided")
+		}
+		// last_verified_tree_size and page_token are allowed to change between paginated requests.
+		// Clear them here both for comparison and for encoding next_page_token in the response.
+		in.LastVerifiedTreeSize = 0
```

Comment:
if you set this to 0 you lose this information which you need. I think you can pass an option to the proto comparison to ignore certain fields.

Reply:
I store it in lastVerified and use that instead.

I also had a vague recollection that you can compare ignoring fields, but I think that's a contrib library (and maybe only in Java / C++).